packages/core/src/com/communication.ts

```diff
@@ -339,6 +339,14 @@ export class Communication {
         return this.environments[envName]?.host;
     }
 
+    public getRegisteredEnvironmentNames(): string[] {
+        return Object.keys(this.environments).filter((id) => id !== '*');
+    }
+
+    public resetDisposedEnvironments(): void {
+        this.disposedEnvironments.clear();
```

Comment:
I agree, but that was added a long time ago in order to avoid subscribing to disposed environments events.

This is supposed to be used in order to reinitialize the application (e.g. like in VS code when you are opening another folder). This can reset all envs except the root one and reinitialize everything from a scratch.